samples/nextjs/scripts/generate-plugin.ts

```diff
@@ -0,0 +1,92 @@
+﻿import fs from 'fs';
+import path from 'path';
+import { getItems } from './utils';
+
+/*
+  PLUGINS GENERATION
+  NOTE: pluginName: the name of the plugin in the src/lib folder
+  Generates the `/src/temp/{pluginName}-plugins.ts` file, which exports list of plugins
+
+  Generating the plugins is optional, and it can be maintained manually if preferred.
+
+  The default convention uses the plugin's filename (without the extension) as the first part of the component
+  name. For example, the file `/lib/page-props-factory/plugins/exampleName.ts` would map to plugin `exampleNamePlugin`.
+  This can be customized in writePlugins().
+
+  This script supports two modes. In default mode, the plugins file is written once.
+  In watch mode, the plugins source folder is watched, and {pluginName}-plugins.ts is
+  regenerated whenever files are added or deleted. Run in watch mode by passing a `--watch` argument
+  when calling the script.
```

Comment:
I don't think this comment is relevant any longer (we decided to drop watch mode support).